pkg/controller/common/rollout/workloads/deployment_scale_controller.go

```diff
@@ -274,56 +259,33 @@ func (s *DeploymentScaleController) Finalize(ctx context.Context, succeed bool)
 		s.rolloutStatus.RolloutRetry(err.Error())
 		return false
 	}
-	deployPatch := client.MergeFrom(s.deploy.DeepCopyObject())
-	// remove the parent controller from the resources' owner list
-	var newOwnerList []metav1.OwnerReference
-	isOwner := false
-	for _, owner := range s.deploy.GetOwnerReferences() {
-		if owner.Kind == v1beta1.AppRolloutKind && owner.APIVersion == v1beta1.SchemeGroupVersion.String() {
-			isOwner = true
-			continue
-		}
-		newOwnerList = append(newOwnerList, owner)
-	}
-	if !isOwner {
-		// nothing to do if we are already not the owner
-		klog.InfoS("the deployment is already released and not controlled by rollout", "deployment", s.deploy.Name)
-		return true
-	}
```

Comment:
why delete?

Reply:
The work would be done by the `releaseDeployment` method.